test/gtest/ucp/test_ucp_wireup.cc

```diff
@@ -922,7 +922,12 @@ class test_ucp_wireup_errh_peer : public test_ucp_wireup_1sided
         skip_loopback();
     }
 
-    static void err_cb(void *, ucp_ep_h, ucs_status_t) {}
+    static void err_cb(void *arg, ucp_ep_h ep, ucs_status_t status)
+    {
+        test_ucp_wireup_errh_peer *self =
+                reinterpret_cast<test_ucp_wireup_errh_peer*>(arg);
+        ucp_test::err_handler_cb(self->find_entity_by_ep(ep), ep, status);
+    }
```

Comment:
how do the changes in gtest test the fix in this PR (that send completion callback is not called before uct ep is destroyed)?

Reply:
I've updated all tests which provide its own error handling callback (instead of default one), because they also need to do `add_err` for entity in order to skip flushing failed EP, since it hangs - we moved purging requests to the place when UCP EP is closed instead of from UCP EP discard lanes callback

Reply:
1. this means that this PR broke the case of "flush after fail". i would expect flush after fail with return some error, and not hang - since all lanes are failed_ep
2. would the new gtests fail if we hadn't moved the reqs purge to ep_destroy_base? i would expect the answer is "yes" which would mean the new tests actually test the fix